src/content/tutorial/ui/6-stateful-widget.md

```diff
@@ -0,0 +1,189 @@
+---
+title: Stateful widgets
+description: Learn about StatefulWidgets and rebuilding Flutter UI.
+permalink: /tutorial/stateful-widget/
+---
+
+{%- comment %} TODO(ewindmill) embed video {%- endcomment %}
+
+So far, your app displays a grid and an input field, but the grid
+doesn't yet update to reflect the user’s guesses. When this app is
+complete, each tile in the next unfilled row should update after each
+submitted user guess by:
+
+* Displaying the correct letter  
+* Changing color to reflect whether the letter is correct (green), is
+  in the word but at an incorrect position (yellow), or incorrect in
+  both position and type (grey).
+
+To handle this dynamic behavior, you need to convert `GamePage` from a
+`StatelessWidget` to a [`StatefulWidget`][].
+
+## Why stateful widgets?
+
+When a widget's appearance or data needs to change during its
+ lifetime, you need a `StatefulWidget` and a companion `State` object.
+While the `StatefulWidget` itself is still immutable (its properties
+can't change after creation), the `State` object is long-lived, can
+hold mutable data, and can be rebuilt when that data changes, causing
+the UI to update.
+
+{%- comment %} TODO(ewindmill) diagram {%- endcomment %}
+
+The basic `StatefulWidget` structure:
+
+```dart
+class ExampleWidget extends StatefulWidget {
+  ExampleWidget({super.key});
+  
+  @override
+  State<ExampleWidget> createState() => _ExampleWidgetState();
+}
+
+class _ExampleWidgetState extends State<ExampleWidget> {
+  @override
+  Widget build(BuildContext context) {
+   return Container();
+  }
+}
+```
+
+## Convert `GamePage` to a stateful widget
+
+To convert the `GamePage` widget (or any other) from a stateless widget to a stateful widget, do the following steps:
+
+1. Change `GamePage` to extend `StatefulWidget` instead of
+   `StatelessWidget`.  
```

Comment:
Wait! Wait! Wait! I distinctly remember that you can do this automatically in an IDE. Is that no longer true?